src/app/pages/Roadmap.tsx

```diff
@@ -85,7 +86,33 @@ export const Roadmap: React.FC = (): JSX.Element => {
     const classes = useStyles();
     const [frameworkFilter, setFrameworkFilter] = useState<FrameworkFilter>('all');
     const [quarterFilter, setQuarterFilter] = useState<Quarter | 'Quarter'>('Quarter');
+    const [roadmap, setRoadmap] = useState<RoadmapBucket[]>([]);
+    const [loading, setLoading] = useState<boolean>(true);
     const theme = useTheme();
+    const loadingGroups = [
+        [1, 2],
+        [1, 2, 3, 4],
+        [1, 2, 3],
+    ];
+
+    useEffect(() => {
```

Comment:
refer to the useEffect in `BugsButton` for returning a cleanup function that will eliminate functions running on unmounted components.

Reply:
I don't understand exactly what is happening here, not sure how BugsButton will guarantee a mount before the API call comes back.  I copied the logic and moved it over the the Roadmap component.  I figure since the call will always be asynchronous then the component should moment even if the content is cached on the client.

Reply:
It's not about guaranteeing a mount before the call comes back (that's guaranteed based on the fact that we don't make the call until we're mounted)...it's more about checking if the component is still mounted before setting state. If you call setRoadmap (or setLoading) and the component is unmounted (e.g. from navigating to another screen before the asynchronous call resolves) you'll get an error/exception in the console (e.g., "cannot set state on an unmounted component"). The asynchronous functions don't get automatically canceled when the component is unmounted, so we have to check ourselves.

It's probably a little overkill, but it's nice to have the same structure for all of our API calls :)